scripts/clones_analysis/clones_features_builder.py

```diff
@@ -0,0 +1,50 @@
+import networkx as nx
+import pandas as pd
+
+
+def add_number_of_clones_features(methods_df: pd.DataFrame, graph: nx.Graph = None):
+    methods_df['n_clones'] = methods_df['method_id'].apply(lambda id: number_of_clones(id, graph))
+    methods_df['n_inter_clones'] = methods_df['method_id'].apply(
+        lambda id: number_of_clones(id, graph, only_inter=True))
+    methods_df['n_100_clones'] = methods_df['method_id'].apply(
+        lambda id: number_of_clones(id, graph, only_100_closeness=True))
+    methods_df['n_inter_100_clones'] = methods_df['method_id'].apply(
+        lambda id: number_of_clones(id, graph, only_inter=True, only_100_closeness=True))
+
+
+def number_of_clones(method_id: int, graph: nx.Graph, only_inter: bool = False,
+                     only_100_closeness: bool = False) -> int:
```

Comment:
I think it is the wrong way to make a lot of bool fields. Maybe should we create a frozen data class with these fields and use an instance of it? In this way, we can add new condition easily

Reply:
Perhaps it would be better to make a separate class for filtering, in which there will be a bool value and a function that takes an edge. And in the arguments, we can use a list of these instances.

```
if not all( map(lamda conf: conf.to_count(edge), conf_list) ):
    continue
```

In this case, we can send different filter config list in the different cases without changing the logic of this function. What do you think?